pkg/istio/authorizationpolicy/controller.go

```diff
@@ -0,0 +1,549 @@
+						if err != nil {
+							log.Errorln("error creating authz policy: ", err.Error())
+							return err
+						}
+						log.Infoln("Revision number is: ", revision)
+					}
+				} else {
+					// case when service has authz flag switch from true to false, authz policy with the same name present
+					if res := c.configStoreCache.Get(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace); res != nil {
+						log.Infoln("Deleting Authz Policy ... ")
+						c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+					}
+				}
+			} else {
+				// case when service has authzEnabledAnnotation removed, and authz policy with the same name present
+				if res := c.configStoreCache.Get(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace); res != nil {
+					log.Infoln("Deleting Authz Policy ... ")
+					c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+				}
+			}
+		} else if castItem.Operation == model.EventDelete {
+			log.Infoln("Deleting Authz Policy ... ")
+			err = c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+		}
+	} else if obj, ok := (castItem.Resource).(*adv1.AthenzDomain); ok {
+		// athenz domain update will result in update in the authz policies in the corresponding namespace
+		if castItem.Operation == model.EventUpdate {
+			// authz policies exist in the namespace
+			res, err := c.configStoreCache.List(schemas.AuthorizationPolicy.Type, m.DomainToNamespace(obj.Name))
+			if err != nil {
+				return fmt.Errorf("Unable to list authz policies in namespace: %s", obj.Namespace)
+			}
+			for _, authzPolicy := range res {
+				signedDomain := obj.Spec.SignedDomain
+				domainRBAC := m.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.adIndexInformer)
+				authzSpec, ok := (authzPolicy.Spec).(*v1beta1.AuthorizationPolicy)
+				if !ok {
+					return fmt.Errorf("unable to cast interface to authorizationpolicies object")
+				}
+				convertedCR := c.convertAthenzModelIntoIstioAuthzPolicy(domainRBAC, authzPolicy.Namespace, authzPolicy.Name, authzSpec.Selector.MatchLabels["svc"])
+				log.Infof("Athenz Domain %s updated, updating Authz Policy in namespace %s ... ", obj.Name, m.DomainToNamespace(obj.Name))
+				// assign current revision, update function requires a defined resource version
+				convertedCR.ResourceVersion = authzPolicy.ResourceVersion
+				revision, err := c.configStoreCache.Update(convertedCR)
+				if err != nil {
+					log.Errorln("error updating authz policy: ", err.Error())
+					return err
+				}
+				log.Infoln("Revision number is: ", revision)
+			}
+		}
+	} else {
+		return fmt.Errorf("unable to cast interface to service or athenzDomain object")
+	}
+	return nil
+}
+
+// checkAuthzEnabledAnnotation checks if current servce object has "authz.istio.io/enabled" annotation set
+func (c *Controller) checkAuthzEnabledAnnotation(serviceObj *corev1.Service) bool {
```

Comment:
It looks like this function is not being used?

Reply:
it is being used to check before authz policy create/update